sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -799,7 +800,11 @@ private void messageReceived(final ChannelHandlerContext context, final RntbdRes
                     break;
 
                 case StatusCodes.REQUEST_TIMEOUT:
-                    cause = new RequestTimeoutException(error, lsn, partitionKeyRangeId, responseHeaders);
+                    Exception inner = new RequestTimeoutException(error, lsn, partitionKeyRangeId, responseHeaders);
```

Comment:
Why need to create RequestTimeoutException as inner? would not 'error' itself enough?

Reply:
To not loose context in cases where we iterate through exceptions recursively through all inner exceptions etc.